bitforge/encoding.py

```diff
@@ -1,68 +1,67 @@
-import utils, binascii, hashlib
-from errors import StringError
+import binascii
+import hashlib
 
+from bitforge import error
+from bitforge.utils import encoding
 
-class EncodingError(StringError):
-    "The string {string} is not properly encoded"
 
+class Error(error.BitforgeError):
+    pass
 
-class InvalidBase58h(EncodingError):
-    "The string {string} is not valid base58/check"
 
+def b2a_hex(data):
+    """Convert a byte buffer to an hexadecimal string."""
 
-class InvalidHex(EncodingError):
-    "The string {string} is not valid hexadecimal"
+    return binascii.b2a_hex(data)
 
 
-def encode_base58h(bytes):
-    return utils.encoding.b2a_hashed_base58(bytes)
+def a2b_hex(string):
+    """Convert an hexadecimal string to a byte buffer."""
 
+    if len(string) % 2 == 1:
+        string = '0' + string
 
-def decode_base58h(string):
     try:
-        return utils.encoding.a2b_hashed_base58(string)
+        return binascii.a2b_hex(string.encode('ascii'))
+    except TypeError:
+        raise Error('Invalid hexadecimal string')
 
-    except utils.encoding.EncodingError:
-        raise InvalidBase58h(string)
 
+def b2i_bigendian(data):
+    """Convert a big endian byte buffer to an unsigned big integer."""
 
-def encode_int(integer, big_endian = True):
-    bytes = bytearray()
+    # Encoding and decoding from hexa appears to be way faster than manually
+    # decoding the buffer in python.
+    return int(b2a_hex(data), 16)
 
-    while integer > 0:
-        bytes.append(integer & 0xff)
-        integer >>= 8
 
-    if big_endian:
-        bytes.reverse()
+def i2b_bigendian(number, num_bytes = 0):
+    """Convert an unsigned big integer to a zero-padded big endian byte buffer.
+    """
 
-    return str(bytes)
+    # Encoding and decoding from hexa appears to be way faster than manually
+    # decoding the buffer in python.
+    return a2b_hex('%0*x' % (2 * num_bytes, number))
```

Comment:
Do you think we should check whether number is a non-negative integer? This is actually a wider discussion: should we check function argument types as a general strategy, like cryptography does, but contrary with the duck typing philosophy?